### PR TITLE
LPS-64753

### PR DESCRIPTION
**Pull request by Ayoub3r**

### CODE REVIEW

**Comment by pedroamarall** [comment]
ci:test:sf

**Comment by pedroamarall** [comment]
> ### ❌ ci:test:sf - 0 out of 1 jobs passed in 5 minutes
> Click here for more details.
> #### Base Branch:
> Branch Name: [master](https://github.com/liferay/liferay-portal/tree/master)Branch GIT ID: [2f31084975b8217be699b430883a9f0a97c1eefc](https://github.com/liferay/liferay-portal/commit/2f31084975b8217be699b430883a9f0a97c1eefc)
> 
> #### Sender Branch:
> Branch Name: [LPS-64753](https://github.com/Ayoub3r/liferay-portal/tree/LPS-64753)Branch GIT ID: [957d457b79a20449a29d2ae0969e5f28510ec85c](https://github.com/Ayoub3r/liferay-portal/commit/957d457b79a20449a29d2ae0969e5f28510ec85c)
> 
> 0 out of 1jobs PASSED
> #### 1 Failed Jobs:
> * [test-portal-source-format](https://test-1-31.liferay.com/job/test-portal-source-format/5788/)
> 
> ##### For more details click [here](https://test-1-31.liferay.com/userContent/jobs/test-portal-source-format/builds/5788/jenkins-report.html).
>      [java] java.lang.Exception: Found 4 formatting issues:
>      [java] 1: Truncated message :
>      [java] /opt/dev/projects/github/liferay-portal/portal-web/test/functional/com/liferay/portalweb/macros/MyMacroFile.macro expected:<definition \{
>      [java] 
>      [java] [  // Access Portal Global Menu and validate CONFIGURATION string at Control Panel
>      [java] 
>      [java]   macro AddFrenchTranslation \{
>      [java]     Form.addTranslation(localizedLanguage = "French (France)");
>      [java] 
>      [java]     Form.editPageTitle(
>      [java]       formPageName = "Fran��ais",
>      [java]       formTitleNumber = 1);
>      [java] 
>      [java]     Click(
>      [java]       key_name = "Full Name",
>      [java]       locator1 = "FormFields#FIELD_TARGET_LABEL");
>      [java] 
>      [java]     FormFields.editText(
>      [java]       fieldName = "label",
>      [java]       fieldValue = "Nom et Pr��nom");
>      [java] 
>      [java]     Click(
>      [java]       key_name = "Age",
>      [java]       locator1 = "FormFields#FIELD_TARGET_LABEL");
>      [java] 
>      [java]     FormFields.editText(
>      [java]       fieldName = "label",
>      [java]       fieldValue = "��ge");
>      [java] 
>      [java]     Form.save();
>      [java]   \}
>      [java] 
>      [java]   macro AddPortugueseTranslation \{
>      [java]     Form.addTranslation(localizedLanguage = "Portuguese (Brazil)");
>      [java] 
>      [java]     Form.editPageTitle(
>      [java]       formPageName = "Portugu��s",
>      [java]       formTitleNumber = 1);
>      [java] 
>      [java]     Click(
>      [java]       key_name = "Full Name",
>      [java]       locator1 = "FormFields#FIELD_TARGET_LABEL");
>      [java] 
>      [java]     FormFields.editText(
>      [java]       fieldName = "label",
>      [java]       fieldValue = "Nome completo");
>      [java] 
>      [java]     Click(
>      [java]       key_name = "Age",
>      [java]       locator1 = "FormFields#FIELD_TARGET_LABEL");
>      [java] 
>      [java]     FormFields.editText(
>      [java]       fieldName = "label",
>      [java]       fieldValue = "Idade");
>      [java] 
>      [java]     Form.save();
>      [java]   \}
>      [java] 
>      [java]   macro AssertLanguageAndFillOutFormEnglish \{
>      [java]     FormViewBuilder.validateFieldLabel(fieldLabel = "Full Name");
>      [java] 
>      [java]     FormFields.editText(
>      [java]       fieldName = "Full Name",
>      [java]       fieldValue = "Ayoub Er-rfife");
>      [java] 
>      [java]     FormViewBuilder.validateFieldLabel(fieldLabel = "Age");
>      [java] 
>      [java]     FormFields.editText(
>      [java]       fieldName = "Age",
>      [java]       fieldValue = 21);
>      [java]

I spoke to you personally and you are aware of what needs to be done. If you need any help, please tell me

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/controlmenu/ControlMenuPreviewPanel.path [hunk 1]
```diff
@@ -35,6 +35,21 @@
 	<td>//div[contains(@class,'simulation-preview')]//iframe</td>
 	<td></td>
 </tr>
+<tr>
+	<td>PREVIEW_CONFIRM</td>
```
Comment (pedroamarall) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/controlmenu/ControlMenuPreviewPanel.path:39]: Hey @Ayoub3r I believe that the title of this path is not well placed, its function is not "preview confirm" but to apply custom size. Can you change this?

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/controlmenu/ControlMenuPreviewPanel.path [hunk 2]
```diff
@@ -35,6 +35,21 @@
...
+	<td>//button[@aria-label='Apply Custom Size']</td>
+	<td></td>
+</tr>
+<tr>
+	<td>PREVIEW_EXIT</td>
```
Comment (pedroamarall) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/controlmenu/ControlMenuPreviewPanel.path:44]: The same case from "preview confirm". you can use like that "BUTTON_EXIT"

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/FormTranslation.testcase [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+		MyMacroFile.AssertLanguageAndFillOutFormFrench();
+
+		Form.submitAgain();
+
+		MyMacroFile.AssertLanguageAndFillOutFormPortuguese();
+
+		Navigator.openURL();
+
+		MyMacroFile.GoToViewEntries();
+	}
```
Comment (pedroamarall) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/FormTranslation.testcase:44]: You can use BDD in your test (Given, When and Then) This helps anyone who picks up and reads your test to understand every part of it

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/SetScreenSize.testcase [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        MyMacroFile.OpenSimulationMenu();
+        
+        AssertTextEquals.assertPartialText(
+            locator1 = "ControlMenuPreviewPanel#PREVIEW_APPLIED_CHANGES",
+			value1 = "Applied"
+        );
+        
+        MyMacroFile.ExitSimulationMenu();
+        
+    }
```
Comment (pedroamarall) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/SetScreenSize.testcase:45]: The same case about the comment from  "CreateAndTranslateForm" test
tip: Don't use the name "MyMacroFile". Use a name that when reading, you can understand which component you are working on

File: portal-web/test/functional/com/liferay/portalweb/tests/mytests/ValidateString.testcase [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    @priority = "5"
+    test ValidateConfigurationStringAtControlPanel {
+
+        MyMacroFile.OpenAppMenu();
+
+        MyMacroFile.NavigateToControlPanel();
+        
+        MyMacroFile.CheckForStringConfiguration();
+    
+    }
```
Comment (pedroamarall) [inline portal-web/test/functional/com/liferay/portalweb/tests/mytests/ValidateString.testcase:27]: The same case about the comment above

Comment (Ayoub3r) [reply portal-web/test/functional/com/liferay/portalweb/tests/mytests/ValidateString.testcase:27]: Hello Pedro, I have read all your comments and appreciate your support.
I am currently working on the extra testcase and when I'm done with it I'll make sure to fix the issues you mentioned!

Comment (pedroamarall) [reply portal-web/test/functional/com/liferay/portalweb/tests/mytests/ValidateString.testcase:27]: Ok @Ayoub3r just mark my name in here when you finished, Thanks for that

Comment (Ayoub3r) [reply portal-web/test/functional/com/liferay/portalweb/tests/mytests/ValidateString.testcase]: Hey @pedroamarall I'm done with all the changes!

Comment (pedroamarall) [reply portal-web/test/functional/com/liferay/portalweb/tests/mytests/ValidateString.testcase]: Nice, I will check now

**Comment by pedroamarall** [comment]
ci:test:sf

**Comment by Ayoub3r** [comment]
@pedroamarall I deleted the file MyMacroFile because I didn't know that when you delete a file locally you need to do the `g add` command again so that the changes apply.

**Comment by Ayoub3r** [comment]
Hey @pedroamarall I just deleted the MyMacroFile because I didn't know that I needed to do the `g add` command again to apply changes

**Comment by pedroamarall** [comment]
ci:test:sf

**Pull request closed without merging**